tests/test_urban_nature_access.py

```diff
@@ -384,3 +389,72 @@ def test_split_greenspace(self):
         admin_vector = None
         admin_layer = None
+
+    def test_split_population(self):
+        """UNA: test split population optional module."""
+        from natcap.invest import urban_nature_access
+
+        args = _build_model_args(self.workspace_dir)
+        del args['results_suffix']
+
+        admin_geom = [
+            shapely.geometry.box(
+                *pygeoprocessing.get_raster_info(
+                    args['lulc_raster_path'])['bounding_box'])]
+        fields = {
+            'pop_female': ogr.OFTReal,
+            'pop_male': ogr.OFTReal,
+        }
+        attributes = [
+            {'pop_female': 0.56, 'pop_male': 0.44}
+        ]
+        pygeoprocessing.shapely_geometry_to_vector(
+            admin_geom, args['admin_unit_vector_path'],
+            pygeoprocessing.get_raster_info(
+                args['population_raster_path'])['projection_wkt'],
+            'GeoJSON', fields, attributes)
+
+        urban_nature_access.execute(args)
+
+        summary_vector = gdal.OpenEx(
+            os.path.join(args['workspace_dir'], 'output', 'admin_units.gpkg'))
+        summary_layer = summary_vector.GetLayer()
+        self.assertEqual(summary_layer.GetFeatureCount(), 1)
+        summary_feature = summary_layer.GetFeature(1)
+
+        def _read_and_sum_raster(path):
+            array = pygeoprocessing.raster_to_numpy_array(path)
+            nodata = pygeoprocessing.get_raster_info(path)['nodata'][0]
+            return numpy.sum(array[~utils.array_equals_nodata(array, nodata)])
+
+        intermediate_dir = os.path.join(args['workspace_dir'], 'intermediate')
+        for (supply_type, supply_field), fieldname in itertools.product(
+                [('over', 'Povr_adm'), ('under', 'Pund_adm')], fields.keys()):
+            groupname = fieldname.replace('pop_', '')
+            supply_raster_path = os.path.join(
+                 intermediate_dir,
+                 f'{supply_type}supplied_population.tif')
+            group_supply_raster_path = os.path.join(
+                 intermediate_dir,
+                 f'{supply_type}supplied_population_{groupname}.tif')
+            pop_proportion = summary_feature.GetField(fieldname)
+            computed_value = summary_feature.GetField(
+                f'{supply_field}_{groupname}')
+
+            numpy.testing.assert_allclose(
+                computed_value,
+                _read_and_sum_raster(supply_raster_path) * pop_proportion,
+                rtol=1e-6
+            )
+            numpy.testing.assert_allclose(
+                computed_value,
+                _read_and_sum_raster(group_supply_raster_path),
+                rtol=1e-6
+            )
+
+        # verify additional rasters exist and have reasonable values relative
+        # to the total rasters
+        #   undersupply of greenspace
+        #   opversupply of greenspace
+        # Also, fields must exist in output vector and have correct values,
+        # whicih should also match the raster.
```

Comment:
Want to keep this comment?

Reply:
Nope!  It's already implemented. Removed in 24f891118